### PR TITLE
Refactoring/modal

### LINKED ISSUE #32: As a user, I should see a central, and prominent add post button that encourages me to click it!

### PR DESCRIPTION
**Pull request by k-atwhite**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Linter has passed locally and any fixes were made for failures
- [x] CircleCI build is passing
- [x] All tests passing locally

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

Issue Number:

## What is the new behavior?

Image Uploader in separate component
Post Button and Bee Cute-ified
Modal Refactored
User Component added and rendered

- closes #32
- loses #45

## Other information

There was a breaking issue before I moved the ImageUploader. I noticed if first while running the cypress test. 
It breaks whenever you click submit, whether adding an image or not. 

After clicking submit, you get this error:

Screen Shot 2021-09-03 at 9 34 50 PM

### CODE REVIEW

**Review by mkrumholz — CHANGES_REQUESTED** [review]
@k-atwhite I pulled the branch and reproduced the error in development. Looks like it's a bad user id (listed currently as 17, but should be somewhere between 21-24) that's hard-coded in. Ready to add auth yet??? Sorry if that was a frustrating error to deal with :(

File: src/components/Header/Header.css [hunk 1]
```diff
@@ -11,11 +11,38 @@ header {
 .logo {
   height: 5em;
   width: 5em;
-  margin-left: 2em;
+  margin-left: 1em;
+}
+
+.logo:hover {
```
Comment (clairefields15) [inline src/components/Header/Header.css:17]: Shaky bee! Love it

File: src/components/ImageUploader/ImageUploader.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect } from 'react';
+import ImageUploading, { ImageListType } from 'react-images-uploading';
+import './ImageUploader.css';
+
+interface ImageUploaderProps {
+  createDirectUpload: ({}) => void;
+}
+
+export const ImageUploader: React.FC<ImageUploaderProps> = ({
```
Comment (clairefields15) [inline src/components/ImageUploader/ImageUploader.tsx:9]: Good work breaking this into it's own component, should make Modal much cleaner!

File: src/components/PostedBy/PostedBy.tsx [hunk 3]
```diff
@@ -12,8 +13,12 @@ export const PostedBy: React.FC<PostedByProps> = ({ post }) => {
   const { avatar, username } = post.user;
 
   return (
-    <div className='user-info'>
-      <img className='avatar' src={avatar} alt={`${username}'s avatar`} />
+    <div className="user-info">
+      <img
+        className="avatar"
+        src={`${baseURL}${avatar}`}
```
Comment (clairefields15) [inline src/components/PostedBy/PostedBy.tsx:19]: Yay!

File: src/components/User/User.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  useEffect(() => {
+    if (!loading && data) {
+      setUser(data.user);
+    }
+  }, [data, loading]);
+
+  return (
+    <>
+      {user && (
```
Comment (clairefields15) [inline src/components/User/User.tsx:20]: We should add a conditional for error here, I think

**Review by clairefields15 — APPROVED** [review]
Looks good. I'll merge and check out the weird error(s)

**Review by k-atwhite — COMMENTED** [review]
Questions resolved! User ID fixed. But now having trouble loading images again...

**Review by mkrumholz — APPROVED** [review]

**Pull request merged**